### PR TITLE
8254782: Fix benchmark issues in java/lang/StringIndexOfChar.java benchmark

### PR DESCRIPTION
**Pull request by jasontatton-aws**
Please review the improvements which I have made to the java/lang/StringIndexOfChar.java jmh benchmark. Please let me know if any further improvements are required.

Thanks,
Jason

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8254782](https://bugs.openjdk.java.net/browse/JDK-8254782): Fix benchmark issues in java/lang/StringIndexOfChar.java benchmark

### Reviewers
 * [Claes Redestad](https://openjdk.java.net/census#redestad) (@cl4es - **Reviewer**)
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/918/head:pull/918`
`$ git checkout pull/918`

### CODE REVIEW

File: test/micro/org/openjdk/bench/java/lang/StringIndexOfChar.java [hunk 1]
```diff
@@ -41,18 +43,37 @@
...
+
+    private Random rng;
+    private String [] latn1_short;
+    private String [] latn1_sse4;
+    private String [] latn1_avx2;
+    private String [] latn1_mixedLength;
+    private String [] utf16_short;
+    private String [] utf16_sse4;
+    private String [] utf16_avx2;
+    private String [] utf16_mixedLength;
```
Comment (cl4es) [inline test/micro/org/openjdk/bench/java/lang/StringIndexOfChar.java:63]: I'd be a little bit happier with `String[]` rather than `String []`

**Review by cl4es — APPROVED** [review]
Looks OK.

FWIW the current looping scheme look like what the [SafeLooping.measureWrong_2](https://github.com/openjdk/jmh/blob/620036ec27db3503fa2dee476036b3219c9a38bc/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_34_SafeLooping.java#L115) warns us about, and could possibly be improved by taking cues from [SafeLooping.measureRight_1](https://github.com/openjdk/jmh/blob/620036ec27db3503fa2dee476036b3219c9a38bc/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_34_SafeLooping.java#L136) and sink each `what.indexOf(..)` into a `BlackHole`. This might matter here since each unit of work is quite small. 

I'll approve the patch as-is, but please consider making those changes.

File: test/micro/org/openjdk/bench/java/lang/StringIndexOfChar.java (hunk 1 above)
Comment (jasontatton-aws) [reply test/micro/org/openjdk/bench/java/lang/StringIndexOfChar.java]: Thanks Claes, I have made the modifications as advised

**Review by shipilev — APPROVED** [review]
Looks fine to me, thanks.

**Review by cl4es — APPROVED** [review]

**Pull request closed without merging**